src/App.tsx

```diff
@@ -1,12 +1,142 @@
 import React from 'react';
 import './App.scss';
+import 'bulma';
+import classNames from 'classnames';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getData } from './api/goods';
+import { GoodsList } from './components/GoodsList';
+import { LoadingError } from './components/LoadingError';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[];
+  isLoading: boolean;
+  hasLoadingError: boolean;
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+    isLoading: false,
+    hasLoadingError: false,
+  };
+
+  loadAllGoods = async () => {
+    this.setState({
+      isLoading: true,
+      hasLoadingError: false,
+    });
+
+    try {
+      const goods = await getData();
+
+      this.setState({
+        goods,
+        isLoading: false,
+      });
+    } catch (error) {
+      this.setState({
+        isLoading: false,
+        hasLoadingError: true,
+      });
+    }
+  };
+
+  load5FirstGoods = async () => {
```

Comment:
Your handlers are similar. You can create just one that get needed callback and call it